Kernel/Graphics/Intel/NativeDisplayConnector.cpp

```diff
@@ -0,0 +1,612 @@
+/*
+ * Copyright (c) 2022, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Bus/PCI/API.h>
+#include <Kernel/Debug.h>
+#include <Kernel/Graphics/Console/ContiguousFramebufferConsole.h>
+#include <Kernel/Graphics/GraphicsManagement.h>
+#include <Kernel/Graphics/Intel/NativeDisplayConnector.h>
+#include <Kernel/Memory/Region.h>
+
+namespace Kernel {
+
+namespace IntelGraphics {
+
+#define DDC2_I2C_ADDRESS 0x50
+
+struct PLLSettings {
+    bool is_valid() const { return (n != 0 && m1 != 0 && m2 != 0 && p1 != 0 && p2 != 0); }
+    u64 compute_dot_clock(u64 refclock) const
+    {
+        return (refclock * (5 * (m1) + (m2)) / (n)) / (p1 * p2);
```

Comment:
The brackets here are a bit strange. I believe this expression is parsed as `((refclock * ((5 * m1) + m2)) / n) / (p1 * p2)`, which is equivalent to `(refclock * (5 * m1 + m2) / n) / (p1 * p2)`. I'm assuming this calculation is what you intend, the brackets here around `m1`, `m2` and `n` are unnecessary.